examples/add/train_add_distill_sd_wds.py

```diff
@@ -0,0 +1,1673 @@
+            pretrained_feature_network=pretrained_feature_network,
+            patch_size=patch_size,
+            hooks=hooks,
+            start_index=start_index,
+        )
+
+        # Trainable discriminator heads
+        heads = []
+        for i in range(self.feature_network.num_hooks):
+            heads += [str(i), DiscriminatorHead(self.feature_network.embed_dim, cond_embedding_dim)]
+        self.heads = torch.nn.ModuleDict(heads)
+
+    def train(self, mode: bool = True):
+        self.feature_network = self.feature_network.train(False)
+        self.heads = self.heads.train(mode)
+        return self
+
+    def eval(self):
+        return self.train(False)
+
+    def forward(
+        self,
+        x: torch.Tensor,
+        c: torch.Tensor,
+        transform_positive=True,
+        return_dict: bool = True,
+    ):
+        # TODO: do we need the augmentations from the original StyleGAN-T code?
+        if transform_positive:
+            # Transform to [0, 1].
+            x = x.add(1).div(2)
+
+        # Forward pass through feature network.
+        features = self.feature_network(x)
+
+        # Apply discriminator heads.
+        logits = []
+        for k, head in self.heads.items():
+            logits.append(head(features[k], c).view(x.size(0), -1))
+        logits = torch.cat(logits, dim=1)
+
+        if not return_dict:
+            return (logits,)
+
+        return DiscriminatorOutput(logits=logits)
+
+
+def log_validation(vae, unet, args, accelerator, weight_dtype, step, name="student"):
+    logger.info("Running validation... ")
+
+    unet = accelerator.unwrap_model(unet)
+    pipeline = StableDiffusionPipeline.from_pretrained(
+        args.pretrained_teacher_model,
+        vae=vae,
+        unet=unet,
+        revision=args.revision,
+        torch_dtype=weight_dtype,
+    )
+    pipeline = pipeline.to(accelerator.device)
+    pipeline.set_progress_bar_config(disable=True)
+
+    if args.enable_xformers_memory_efficient_attention:
+        pipeline.enable_xformers_memory_efficient_attention()
+
+    if args.seed is None:
+        generator = None
+    else:
+        generator = torch.Generator(device=accelerator.device).manual_seed(args.seed)
+
+    validation_prompts = [
+        "portrait photo of a girl, photograph, highly detailed face, depth of field, moody light, golden hour, style by Dan Winters, Russell James, Steve McCurry, centered, extremely detailed, Nikon D850, award winning photography",
+        "Self-portrait oil painting, a beautiful cyborg with golden hair, 8k",
+        "Astronaut in a jungle, cold color palette, muted colors, detailed, 8k",
+        "A photo of beautiful mountain with realistic sunset and blue lake, highly detailed, masterpiece",
+    ]
+
+    image_logs = []
+
+    for _, prompt in enumerate(validation_prompts):
+        images = []
+        with torch.autocast("cuda"):
+            images = pipeline(
+                prompt=prompt,
+                num_inference_steps=4,
```

Comment:
should default to `1` since that's what we're aiming for here.

Reply:
I have set `num_inference_steps` to 1.